### ISSUE TITLE
Specify IBRION with calc_minimize() for VASP jobs

### ISSUE THREAD

**Issue by prince-mathews** [pyiron/pyiron_atomistics#384]
**Summary**

I would like to suggest adding an argument for IBRION in `calc_minimize()` with Vasp jobs. This would allow the user to choose the ionic relaxation algorithm instead of using the default as IBRION=2.

**Detailed Description**

For ionic relaxation with Vasp, the available choices are quasi Newton algorithm (IBRION=1) or conjugate gradient algorithm (IBRION=2) or damped molecular dynamics (IBRION=3). I was recently running some jobs with ionic relaxation and I wanted to use IBRION=1, but since the default with `calc_minimize()` is IBRION=2, I had to specify it separately as `job.input.incar['IBRION]=1`. Adding an argument for IBRION would allow changing the value from within `calc_minimize()` and 
Vasp specific commands can be avoided.

**Further Information, Files, and Links**

**Comment by samwaseda** [comment]
More generic would be something like `method='CG'` or `method='QN'`, so ideally it should be something like:

```python
job = pr.create.job.Vasp('vasp')
...
job.calc_minimize(method='CG')
job.run()
```

**Comment by samwaseda** [comment]
Alternatively, we can slowly start thinking about creating a more modular structure (which should be compatible with @liamhuber's plug-in idea). One possible interface, which would be probably the closest to what we have, could be something like this:

```python
vasp = pr.create.job.Vasp('vasp')
...
cg = vasp.create.job.ConjugateGradient()
cg.run()
```

This one would allow for a clearer overview over the parameters available, which should be accessible from `cg.input`, because some minimisers require different parameters than others.

Well, this being said probably @liamhuber will disagree here because as far as I know he wants a **plugin** and not another layer of job hierarchy, which I personally favour.

**Comment by pmrv** [comment]
I agree it's a bit annoying to have to set it manually in a VASP specific way.   Still I'd hesitate to make into an argument for `calc_*` until we verified that we can do so in a way that works for all codes and not just something that we support for VASP.

**Comment by liamhuber** [comment]
> I agree it's a bit annoying to have to set it manually in a VASP specific way. Still I'd hesitate to make into an argument for `calc_*` until we verified that we can do so in a way that works for all codes and not just something that we support for VASP.

Good ideas from all sides, but I think this one wins it. @samwaseda is right that if/when we add this we want to use a non-code-specific term, but that's exactly because it should work for more than one code! I agree `job.input.incar['IBRION'] = 1` is both ugly and annoying, but since the workaround is  a single, stable and reliable notebook line I'm disinclined to break design philosophy to avoid it...

**Comment by samwaseda** [comment]
It works also for SPHInX if that's your concern.

**Comment by liamhuber** [comment]
> It works also for SPHInX if that's your concern.

I'm not totally opposed to something like you suggested above `job.calc_minimize(method='CG')`. If you think it's plausible to do for a handful of codes go ahead. I guess since it's for ionic relaxation it should work for Vasp, Sphinx, and Lammps at a minimum, or we shouldn't do it.

**Comment by samwaseda** [comment]
I don't really understand what makes it so important that the functionality exists everywhere. If the choice between Quasi Newton and conjugate gradient doesn't exist for LAMMPS it simply issues a warning saying "We don't have QN/CG so we switch back to CG/QN" or an error. Besides if `calc_*` has to exist for all codes, then we have to abolish `calc_md` because it doesn't exist for SPHInX.

**Comment by sudarsan-surendralal** [comment]
> 
> 
> I don't really understand what makes it so important that the functionality exists everywhere. If the choice between Quasi Newton and conjugate gradient doesn't exist for LAMMPS it simply issues a warning saying "We don't have QN/CG so we switch back to CG/QN" or an error. Besides if `calc_*` has to exist for all codes, then we have to abolish `calc_md` because it doesn't exist for SPHInX.

I second this. The only thing we need to implement is a record of all possible (known) optimizers for a given code (accessed by for e.g.: `job_vasp.optimizers.cg`). Then we just need to specify `job_vasp.calc_minimize(optimizer=job_vasp.optimizers.cg)` or `job_vasp.calc_minimize(optimizer="CG")`.

**Comment by liamhuber** [comment]
Indeed, like I said, I'm not totally opposed to it. One of our promises is this abstraction from the underlying codes though, so in cases where a mode doesn't exist (e.g. no `calc_md` for sphinx, `calc_vcsgc` only for Lammps) that's fine. But if I'm a Vasp user and and X user, and I know that both allow CG or QN minimization, and I know that I can call `vasp_job.calc_minimze(method='QN')` then I'm going to be pissed if I can't call `x_job.calc_minimize(method='QN')`. So if you're going to implement such a binding for Vasp, please implement it for the other codes at the same time.

**Comment by pmrv** [comment]
It's not important to me that all codes support all optimizers, but at least all codes need to support to set *an* optimizer.  If it were only possible for VASP there's no point in exposing it in the generic interface if the functionality is not actually generic.  I think @sudarsan-surendralal last suggestion should work nicely.



> Besides if `calc_*` has to exist for all codes, then we have to abolish `calc_md` because it doesn't exist for SPHInX.

Which is actually a good argument to me why `calc_md` should not be defined on `GenericAtomisticJob`.  Codes that do support MD, should be all means have it, though.  (I'm not actually saying to do this now, but since we're making general arguments).
